indexing-service/src/test/java/io/druid/indexing/overlord/ForkingTaskRunnerTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexing.overlord;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterators;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ForkingTaskRunnerTest
+{
+  @Test(expected = AssertionError.class)
+  public void testPatternMatcherFailureForJavaOptions()
+  {
+    checkValues(new String[]{"not quoted has space"});
+  }
```

Comment:
it is non trivial to understand why this throws AssertionError and testPatternMatcherPreservesNonBreakingSpacesJavaOptions() does not. can you pls add some more comments?